rocket_packet.c

```diff
@@ -88,17 +91,63 @@ unsigned int serialize_command_packet(CommandPacket* pkt, uint8_t* dst) {
 unsigned int unpack_command_packet(CommandPacket* pkt, uint8_t* src) {
 	unsigned int offset = 0;
 
-	memcpy((void *) &pkt->start_char, src + offset, 2);
-	offset += 2;
+	memcpy((void *) &pkt->start_short, src + offset, sizeof(pkt->start_short));
+	offset += sizeof(pkt->start_short);
+
+	memcpy((void *) &pkt->id, src + offset, sizeof(pkt->id));
+	offset += sizeof(pkt->id);
+
+	memcpy((void *) &pkt->function, src + offset, sizeof(pkt->function));
+	offset += sizeof(pkt->function);
+
+	memcpy((void *) &pkt->arg, src + offset, sizeof(pkt->arg));
+	offset += sizeof(pkt->arg);
+
+	memcpy((void *) &pkt->crc, src + offset, sizeof(pkt->crc));
+	offset += sizeof(pkt->crc);
+
+	return offset;
+}
+
+unsigned int serialize_ack_packet(AckPacket* pkt, uint8_t* dst) {
+	unsigned int offset = 0;
+	
+	memcpy(dst + offset, (void *) &pkt->start_short, sizeof(pkt->start_short));
+	offset += sizeof(pkt->start_short);
+
+	memcpy(dst + offset, (void *) &pkt->id, sizeof(pkt->id));
+	offset += sizeof(pkt->id);
+	
+	memcpy(dst + offset, (void *) &pkt->ack, sizeof(pkt->ack));
+	offset += sizeof(pkt->ack);
+	
+	memcpy(dst + offset, (void *) &pkt->nack, sizeof(pkt->nack));
+	offset += sizeof(pkt->nack);
+	
+	memcpy(dst + offset, (void *) &pkt->crc, sizeof(pkt->crc));
+	offset += sizeof(pkt->crc);
+	
+	return offset;
+}
+
+
+unsigned int unpack_ack_packet(AckPacket* pkt, uint8_t* src) {
```

Comment:
si tu veux changer tous les noms de fonction pour pack/unpack, go for it, je pense que c'est plus clair que serialize, ou du moins ça prend moins de place à écrire

Reply:
Did I change the name of the function? I think it was serialize when writing to the buffer and unpack when writing in the data structure.